### ISSUE TITLE
[1.18.1] Patchouli Bug

### ISSUE THREAD

**Issue by itsdinkd** [Sirttas/ElementalCraft#69]
Latest forge loader, EC, and patchouli

i am getting this in my logs. please offer compatibility again with patchouli, without this fixed, you get "ERROR NO CONTENT" in all entries. 

```java
[11:25:44] [Render thread/ERROR]: Error compiling book elementalcraft:element_book, using empty contents
java.lang.RuntimeException: Error building entry elementalcraft:instruments/firefurnace
	at vazkii.patchouli.client.book.BookContentsBuilder.lambda$build$2(BookContentsBuilder.java:88) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at java.util.HashMap$Values.forEach(HashMap.java:1065) ~[?:?]
	at vazkii.patchouli.client.book.BookContentsBuilder.build(BookContentsBuilder.java:84) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.common.book.Book.reloadContents(Book.java:190) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.common.book.BookRegistry.reloadContents(BookRegistry.java:164) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.client.book.ClientBookRegistry.reload(ClientBookRegistry.java:63) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.client.base.ClientAdvancements.onClientPacket(ClientAdvancements.java:34) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at net.minecraft.client.multiplayer.ClientAdvancements.handler$zia000$patchouli_onSync(ClientAdvancements.java:1017) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.client.multiplayer.ClientAdvancements.m_104399_(ClientAdvancements.java:57) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.client.multiplayer.ClientPacketListener.m_5498_(ClientPacketListener.java:1127) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateAdvancementsPacket.m_5797_(ClientboundUpdateAdvancementsPacket.java:51) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateAdvancementsPacket.m_5797_(ClientboundUpdateAdvancementsPacket.java:16) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.network.protocol.PacketUtils.m_131356_(PacketUtils.java:21) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:151) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:125) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:110) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1009) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:660) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at net.minecraft.client.main.Main.main(Main.java:205) ~[client-1.18.1-20211210.034407-srg.jar%23303!:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:45) ~[fmlloader-1.18.1-39.0.7.jar%2323!:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:90) [bootstraplauncher-0.1.17.jar:?]
Caused by: java.lang.RuntimeException: Error while loading entry elementalcraft:instruments/firefurnace page 2
	at vazkii.patchouli.client.book.BookEntry.build(BookEntry.java:224) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.client.book.BookContentsBuilder.lambda$build$2(BookContentsBuilder.java:86) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	... 32 more
Caused by: java.lang.IllegalArgumentException: Failure parsing state matcher
	at vazkii.patchouli.common.multiblock.SerializedMultiblock.deserializeMapping(SerializedMultiblock.java:41) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.common.multiblock.SerializedMultiblock.deserializeDense(SerializedMultiblock.java:93) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.common.multiblock.SerializedMultiblock.toMultiblock(SerializedMultiblock.java:103) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.client.book.page.PageMultiblock.build(PageMultiblock.java:72) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.client.book.BookEntry.build(BookEntry.java:221) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	at vazkii.patchouli.client.book.BookContentsBuilder.lambda$build$2(BookContentsBuilder.java:86) ~[Patchouli-1.18.1-61.jar%23223!:1.18.1-61]
	... 32 more
Caused by: com.mojang.brigadier.exceptions.CommandSyntaxException: Unknown block type 'elementalcraft:tank' at position 0: <--[HERE]
```

**Comment by remclave** [comment]
@Sirttas, if it helps with troubleshooting the problem, here's a screenshot of the error...
![Elementopedia](https://user-images.githubusercontent.com/31051239/147857040-73f6ad8c-2233-464a-aaae-61f0490785b1.png)

**Comment by Sirttas** [comment]
fixed in 4.1.7
